nixos/modules/services/web-apps/funkwhale.nix

```diff
@@ -0,0 +1,543 @@
+                alias = "${pkgs.funkwhale}/front/embed.html";
+                extraConfig = ''
+                  add_header Content-Security-Policy "default-src 'self'; script-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; font-src 'self' data:; object-src 'none'; media-src 'self' data:";
+                  add_header Referrer-Policy "strict-origin-when-cross-origin";
+                  add_header X-Frame-Options "ALLOW";
+                  expires 30d;
+                  add_header Pragma public;
+                  add_header Cache-Control "public, must-revalidate, proxy-revalidate";
+                '';
+              };
+              "/federation/" = { 
+                extraConfig = proxyConfig;
+                proxyPass = "http://funkwhale-api/federation/";
+              };
+              "/rest/" = { 
+                extraConfig = proxyConfig;
+                proxyPass = "http://funkwhale-api/api/subsonic/rest/";
+              };
+              "/.well-known/" = { 
+                extraConfig = proxyConfig;
+                proxyPass = "http://funkwhale-api/.well-known/";
+              };
+              "/media/".alias = "${cfg.api.mediaRoot}/";
+              "/_protected/media/" = {
+                extraConfig = ''
+                  internal;
+                '';
+                alias = "${cfg.api.mediaRoot}/";
+              };
+              "/_protected/music/" = {
+                extraConfig = ''
+                  internal;
+                '';
+                alias = "${cfg.musicDirectoryPath}/";
+              };
+              "/staticfiles/".alias = "${cfg.api.staticRoot}/";
+            };
+          };
+        };
+      };
+
+      systemd.tmpfiles.rules = [
+        "d ${cfg.dataDir} 0755 ${cfg.user} ${cfg.group} - -"
+        "d ${cfg.api.mediaRoot} 0755 ${cfg.user} ${cfg.group} - -"
+        "d ${cfg.api.staticRoot} 0755 ${cfg.user} ${cfg.group} - -"
+        "d ${cfg.musicDirectoryPath} 0755 ${cfg.user} ${cfg.group} - -"
+      ];
+
+      systemd.targets.funkwhale = {
+        description = "Funkwhale";
+        wants = ["funkwhale-server.service" "funkwhale-worker.service" "funkwhale-beat.service"];
+      }; 
+      systemd.services = 
+      let serviceConfig = {
+        User = "${cfg.user}";
+        WorkingDirectory = "${pkgs.funkwhale}";
+        EnvironmentFile =  "${funkwhaleEnvFile}";
+      };
+      in {
+        funkwhale-psql-init = mkIf cfg.database.createLocally {
+          description = "Funkwhale database preparation";
+          after = [ "redis.service" "postgresql.service" ];
+          wantedBy = [ "funkwhale-init.service" ];
+          before   = [ "funkwhale-init.service" ];
+          serviceConfig = {
+            User = "postgres";
+            ExecStart = '' ${config.services.postgresql.package}/bin/psql -d ${cfg.database.name}  -c 'CREATE EXTENSION IF NOT EXISTS "unaccent";CREATE EXTENSION IF NOT EXISTS "citext";' '';
```

Comment:
Split this into more lines, same for other long lines, try to keep them to ~80 chars wide